### ISSUE TITLE
Docker build is failing

### ISSUE THREAD

**Issue by inviscid** [greenplum-db/greenplum-for-kubernetes#1]
I added a Github action to fire up the Docker build directly.  This is built on a fork, thus the PlaidCloud name in the paths.

The main culprit seems to be this error: `not enough arguments in call to k8scsr.WaitForCertificate`

The build process progresses to a point but the following error is raised:

```
#15 55.45 # github.com/pivotal/greenplum-for-kubernetes/greenplum-operator/pkg/admission
[289](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:293)
#15 55.45 ../../pkg/admission/certificate.go:96:108: not enough arguments in call to k8scsr.WaitForCertificate
[290](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:294)
#15 55.45 	have (context.Context, "k8s.io/client-go/kubernetes/typed/certificates/v1beta1".CertificateSigningRequestInterface, *"k8s.io/api/certificates/v1beta1".CertificateSigningRequest)
[291](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:295)
#15 55.45 	want (context.Context, kubernetes.Interface, string, types.UID)
[292](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:296)
#15 ERROR: executor failed running [/bin/sh -c cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build]: exit code: 2
[293](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:297)
------
[294](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:298)
 > [build-in-docker 6/6] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod     cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build:
[295](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:299)
#15 3.854 go: downloading github.com/go-openapi/jsonreference v0.19.6
[296](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:300)
#15 3.860 go: downloading github.com/mailru/easyjson v0.7.6
[297](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:301)
#15 3.882 go: downloading github.com/PuerkitoBio/purell v1.1.1
[298](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:302)
#15 3.882 go: downloading github.com/go-openapi/jsonpointer v0.19.5
[299](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:303)
#15 3.894 go: downloading github.com/josharian/intern v1.0.0
[300](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:304)
#15 3.895 go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
[301](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:305)
#15 55.45 # github.com/pivotal/greenplum-for-kubernetes/greenplum-operator/pkg/admission
[302](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:306)
#15 55.45 ../../pkg/admission/certificate.go:96:108: not enough arguments in call to k8scsr.WaitForCertificate
[303](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:307)
#15 55.45 	have (context.Context, "k8s.io/client-go/kubernetes/typed/certificates/v1beta1".CertificateSigningRequestInterface, *"k8s.io/api/certificates/v1beta1".CertificateSigningRequest)
[304](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:308)
#15 55.45 	want (context.Context, kubernetes.Interface, string, types.UID)
[305](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:309)
------
[306](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:310)
ERROR: failed to solve: executor failed running [/bin/sh -c cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build]: exit code: 2
[307](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:311)
Error: buildx failed with: ERROR: failed to solve: executor failed running [/bin/sh -c cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build]: exit code: 2#15 55.45 # github.com/pivotal/greenplum-for-kubernetes/greenplum-operator/pkg/admission
[289](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:293)
#15 55.45 ../../pkg/admission/certificate.go:96:108: not enough arguments in call to k8scsr.WaitForCertificate
[290](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:294)
#15 55.45 	have (context.Context, "k8s.io/client-go/kubernetes/typed/certificates/v1beta1".CertificateSigningRequestInterface, *"k8s.io/api/certificates/v1beta1".CertificateSigningRequest)
[291](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:295)
#15 55.45 	want (context.Context, kubernetes.Interface, string, types.UID)
[292](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:296)
#15 ERROR: executor failed running [/bin/sh -c cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build]: exit code: 2
[293](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:297)
------
[294](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:298)
 > [build-in-docker 6/6] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod     cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build:
[295](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:299)
#15 3.854 go: downloading github.com/go-openapi/jsonreference v0.19.6
[296](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:300)
#15 3.860 go: downloading github.com/mailru/easyjson v0.7.6
[297](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:301)
#15 3.882 go: downloading github.com/PuerkitoBio/purell v1.1.1
[298](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:302)
#15 3.882 go: downloading github.com/go-openapi/jsonpointer v0.19.5
[299](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:303)
#15 3.894 go: downloading github.com/josharian/intern v1.0.0
[300](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:304)
#15 3.895 go: downloading github.com/PuerkitoBio/urlesc v0.0.0-20170810143723-de5bf2ad4578
[301](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:305)
#15 55.45 # github.com/pivotal/greenplum-for-kubernetes/greenplum-operator/pkg/admission
[302](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:306)
#15 55.45 ../../pkg/admission/certificate.go:96:108: not enough arguments in call to k8scsr.WaitForCertificate
[303](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:307)
#15 55.45 	have (context.Context, "k8s.io/client-go/kubernetes/typed/certificates/v1beta1".CertificateSigningRequestInterface, *"k8s.io/api/certificates/v1beta1".CertificateSigningRequest)
[304](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:308)
#15 55.45 	want (context.Context, kubernetes.Interface, string, types.UID)
[305](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:309)
------
[306](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:310)
ERROR: failed to solve: executor failed running [/bin/sh -c cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build]: exit code: 2
[307](https://github.com/PlaidCloud/greenplum-for-kubernetes/actions/runs/4009969312/jobs/6885935975#step:5:311)
Error: buildx failed with: ERROR: failed to solve: executor failed running [/bin/sh -c cd /greenplum-for-kubernetes/greenplum-operator/cmd/greenplumOperator && go build]: exit code: 2
```

**Comment by khuddlefish** [comment]
Thanks for the issue. It's a known issue caused from upgrading K8s versions recently (from 1.17 which is out of support to 1.25) but not quite every issue from that upgrade has been fixed yet. We are hoping to replace our certificate process with `cert-manager`. If you'd like to help with that work, we'd love having your contribution.

**Comment by inviscid** [comment]
Thanks.  We will take a look at it and see if we can get that patched up.

**Comment by feliz13** [comment]
hi, I encountered the same situation. Is there a way to bypass it (such as rolling back to the old code for compilation)?

**Comment by inviscid** [comment]
@feliz13 we are working on a complete build pipeline starting with the community version of Greenplum, adding in PXF, and then finally the GP4K parts.  It isn't quite ready yet but hopefully soon.

In response to your question, we did not opt to use cert-manager as it was a little hard to figure out exactly where changes were necessary.  Instead, we simply updated the references from `v1beta1` to the latest API.
